src/shared/utils/uidl-utils.ts

```diff
@@ -82,82 +82,124 @@ export const traverseNodes = (node: ContentNode, fn: (node: ContentNode) => void
   }
 }
 
-export const splitDynamicAndStaticStyles = (style: StyleDefinitions) => {
-  const staticStyles: StyleDefinitions = {}
-  const dynamicStyles: StyleDefinitions = {}
+interface SplitResponse {
+  staticStyles: UIDLStyleDefinitions
+  dynamicStyles: UIDLStyleDefinitions
+}
+export const splitDynamicAndStaticStyles = (style: UIDLStyleDefinitions): SplitResponse => {
+  // const staticStyles: UIDLStyleDefinitions = {}
+  // const dynamicStyles: UIDLStyleDefinitions = {}
 
-  Object.keys(style).forEach((key) => {
-    const value = style[key]
+  const responsePayload: SplitResponse = { staticStyles: {}, dynamicStyles: {} }
 
-    if (typeof value === 'object') {
-      const nestedResult = splitDynamicAndStaticStyles(value)
-      if (Object.keys(nestedResult.dynamicStyles).length > 0) {
-        dynamicStyles[key] = nestedResult.dynamicStyles
-      }
-      if (Object.keys(nestedResult.staticStyles).length > 0) {
-        staticStyles[key] = nestedResult.staticStyles
-      }
-    } else if (typeof value === 'string' && value.startsWith('$props.')) {
-      dynamicStyles[key] = value
-    } else {
-      staticStyles[key] = value
+  Object.keys(style).reduce((acc: SplitResponse, styleKey) => {
```

Comment:
since we're returning two separate structures, does it help that we use reduce here? a forEach seems more straight forward to parse the style object